specs/waku/waku-1.md

```diff
@@ -104,12 +104,17 @@ Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) w
 ; Packet codes 0 - 127 are reserved for Waku protocol
 packet-code = 1*3DIGIT
 
-; rate limits
-limit-ip     = 1*DIGIT
-limit-peerid = 1*DIGIT
-limit-topic  = 1*DIGIT
+; packet rate limits per second
+packet-limit-ip     = 1*DIGIT
+packet-limit-peerid = 1*DIGIT
+packet-limit-topic  = 1*DIGIT
 
-rate-limits = "[" limit-ip limit-peerid limit-topic "]"
+; rate limits by size in bytes
+bytes-limit-ip     = 1*DIGIT
+bytes-limit-peerid = 1*DIGIT
+bytes-limit-topic  = 1*DIGIT
+
+rate-limits = "[" packet-limit-ip packet-limit-peerid packet-limit-topic bytes-limit-ip bytes-limit-peerid bytes-limit-topic "]"
```

Comment:
Would it not make more sense to use a new field in the `options` alist here?

Appending to the `rate-limits` will give similar issues that we had with the original status messages before the alist was introduced. The last 3 items must become optional basically to provide compatibility. And you can't provide the last 3 without the first 3, making them no longer optional...sort of.
It is not as crucial as the original switch, as it doesn't block all the future general additions, but only the ones for rate limiting, but still I find it unneeded to add this complexity, especially as we have the `options` alist there for a reason.

Reply:
Would be inclined to agree with above

Reply:
Updated